src/GraphNodeManagerViewExtension/ViewModels/GridNodeViewModel.cs

```diff
@@ -305,8 +380,19 @@ private bool IsNodeEmptyList(MirrorData mirrorData)
         {
             if (mirrorData == null || !mirrorData.IsCollection) return false;
 
-            var list = mirrorData.GetElements();
-            return !list.Any();
+            try
+            {
+                var list = mirrorData.GetElements();
+                return !list.Any();
+            }
+            catch(NullReferenceException)
+            {
+                return false;
+            }
+            catch (Exception)
+            {
+                return false;
+            }
```

Comment:
I think this entire function makes sense to expose on `MirrorData` itself.

Reply:
Sorry, @aparajit-pratap , I didn't quite get what you meant there. I have implemented some of the modifications based on your comments from above, could you tell me what you think should happen next?